google/cloud/forseti/scanner/scanners/retention_scanner.py

```diff
@@ -0,0 +1,132 @@
+            scanner_configs (dict): Scanner configurations.
+            service_config (ServiceConfig): Forseti 2.0 service configs
+            model_name (str): name of the data model
+            snapshot_timestamp (str): Timestamp, formatted as YYYYMMDDTHHMMSSZ.
+            rules (str): Fully-qualified path and filename of the rules file.
+        """
+        super(RetentionScanner, self).__init__(
+            global_configs,
+            scanner_configs,
+            service_config,
+            model_name,
+            snapshot_timestamp,
+            rules)
+        self.rules_engine = rre.RetentionRulesEngine(
+            rules_file_path=self.rules,
+            snapshot_timestamp=self.snapshot_timestamp)
+        self.rules_engine.build_rule_book(self.global_configs)
+
+    @staticmethod
+    def _flatten_violations(violations):
+        """Flatten RuleViolations into a dict for each RuleViolation member.
+
+        Args:
+            violations (list): The RuleViolations to flatten.
+
+        Yields:
+            dict: Iterator of RuleViolations as a dict per member.
+        """
+        for violation in violations:
+            yield {
+                'resource_name': violation.resource_name,
+                'resource_id': violation.resource_id,
+                'resource_type': violation.resource_type,
+                'full_name': violation.full_name,
+                'rule_index': violation.rule_index,
+                'rule_name': violation.rule_name,
+                'violation_type': violation.violation_type,
+                'violation_data': violation.violation_data,
+                'resource_data': violation.resource_data
+            }
+
+    def _output_results(self, all_violations):
+        """Output results.
+
+        Args:
+            all_violations (list): All violations
+        """
+        all_violations = self._flatten_violations(all_violations)
+        self._output_results_to_db(all_violations)
+
+    def _find_violations(self, retention_info):
+        """Find violations in the resources with retention policies.
+
+        Args:
+            retention_info (list): Contains all supported resource in retention
+
+        Returns:
+            list: All violations.
+        """
+        LOGGER.info('Finding retention violations...')
+        all_violations = []
+
+        for resource in retention_info:
+            violations = self.rules_engine.find_violations(resource)
+            LOGGER.debug(violations)
+            all_violations.extend(violations)
+
+        return all_violations
+
+    def _retrieve(self):
+        """Retrieves the data for scanner.
+
+        Returns:
+            list: a list of Resources, with a type in
+                SUPPORTED_RETENTION_RES_TYPES
+        """
+        model_manager = self.service_config.model_manager
+        scoped_session, data_access = model_manager.get(self.model_name)
+        retention_info = []
```

Comment:
this should probably be called resources

Reply:
I change it to retention_res. Because I think the return value should be something relevant to the scanner's name based on other scanners.